dwave/optimization/symbols.pyx

```diff
@@ -794,6 +802,54 @@ cdef class Constant(ArraySymbol):
         buffer.strides = <Py_ssize_t*>(self.ptr.strides().data())
         buffer.suboffsets = NULL
 
+    def __index__(self):
+        if not self._is_integer():
+            # Follow NumPy's error message
+            # https://github.com/numpy/numpy/blob/66e1e3/numpy/_core/src/multiarray/number.c#L833
+            raise TypeError("only integer scalar constants can be converted to a scalar index")
+
+        return <Py_ssize_t>deref(self.ptr.buff())
+
+    def __richcmp__(self, rhs, int op):
+        # __richcmp__ is a special Cython method
+
+        # If rhs is another Symbol, defer to ArraySymbol to handle the
+        # operation. Which may or may not actually be implemented.
+        # Otherwise, defer to NumPy.
+        # We could also check if rhs is another Constant and handle that differently,
+        # but that might lead to confusing behavior so we treat other Constants the
+        # same as any other symbol.
```

Comment:
In what was should/could it be handled differently (unless not deferring to NumPy at all)? NumPy would just use `rhs` as if a NumPy array either way since it's being deferred there.

Reply:
> We could also check if rhs is another Constant and handle that differently

Just responding to this comment. Curious if there's anything that should or could be handled differently.

Reply:
Oh. The alternative would be, when both sides of the operation are `Constant`s, to have the result be another `Constant` or even to "demote" to a NumPy array.